azurerm/resource_arm_servicebus_topic_authorization_rule.go

```diff
@@ -0,0 +1,240 @@
+	resGroup := d.Get("resource_group_name").(string)
+
+	rights, err := expandServiceBusTopicAuthorizationRuleAccessRights(d)
+	if err != nil {
+		return err
+	}
+
+	parameters := servicebus.SBAuthorizationRule{
+		Name: &name,
+		SBAuthorizationRuleProperties: &servicebus.SBAuthorizationRuleProperties{
+			Rights: rights,
+		},
+	}
+
+	_, err = client.CreateOrUpdateAuthorizationRule(ctx, resGroup, namespaceName, topicName, name, parameters)
+	if err != nil {
+		return err
+	}
+
+	read, err := client.GetAuthorizationRule(ctx, resGroup, namespaceName, topicName, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read ServiceBus Topic Authorization Rule %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmServiceBusTopicAuthorizationRuleRead(d, meta)
+}
+
+func resourceArmServiceBusTopicAuthorizationRuleRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).serviceBusTopicsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	namespaceName := id.Path["namespaces"]
+	topicsName := id.Path["topics"]
+	name := id.Path["authorizationRules"]
+
+	resp, err := client.GetAuthorizationRule(ctx, resGroup, namespaceName, topicsName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request on Azure ServiceBus Topic Authorization Rule %s: %+v", name, err)
+	}
+
+	keysResp, err := client.ListKeys(ctx, resGroup, namespaceName, topicsName, name)
+	if err != nil {
+		return fmt.Errorf("Error making Read request on Azure ServiceBus Topic Authorization Rule List Keys %s: %+v", name, err)
+	}
+
+	d.Set("name", name)
+	d.Set("topic_name", topicsName)
+	d.Set("namespace_name", namespaceName)
+	d.Set("resource_group_name", resGroup)
+
+	flattenServiceBusTopicAuthorizationRuleAccessRights(d, resp)
+
+	d.Set("primary_key", keysResp.PrimaryKey)
+	d.Set("primary_connection_string", keysResp.PrimaryConnectionString)
+	d.Set("secondary_key", keysResp.SecondaryKey)
+	d.Set("secondary_connection_string", keysResp.SecondaryConnectionString)
+
+	return nil
+}
+
+func resourceArmServiceBusTopicAuthorizationRuleDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).serviceBusTopicsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := parseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resGroup := id.ResourceGroup
+	namespaceName := id.Path["namespaces"]
+	topicName := id.Path["topics"]
+	name := id.Path["authorizationRules"]
+
+	resp, err := client.DeleteAuthorizationRule(ctx, resGroup, namespaceName, topicName, name)
+
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("Error issuing Azure ARM delete request of ServiceBus Topic Authorization Rule '%s': %+v", name, err)
+	}
```

Comment:
checking the SDK here, the method above can return either a 200 or a 204 - can we change this to:

```
if err != nil {
  return fmt.Errorf("Error issuing Azure ARM delete request of ServiceBus Topic Authorization Rule %q (Resource Group %q): %+v", name, resGroup, err)
}
```